File: BOJ/3231.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
int n,cnt;
int arr[100001];
int idx[100001];
int main() {
	scanf("%d", &n);
	for (int i = 1; i <= n; i++) {
		scanf("%d", &arr[i]);
		idx[arr[i]] = i;
	}
	for (int i = 1; i <= n-1; i++){
		int index1 = idx[i];
		int index2 = idx[i + 1];
		if (index1 < index2) continue;
		else if (index1 > index2) cnt++;
	}
	printf("%d\n", cnt);
	return 0;
}

File: BOJ(Java 공부용)/3231.java
import java.io.*;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n + 1];
        int[] idx = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            arr[i] = Integer.parseInt(br.readLine());
            idx[arr[i]] = i;
        }
        int cnt = 0;
        for (int i = 1; i < n; i++) {
            int index = idx[i];
            int index2 = idx[i + 1];
            if (index > index2) cnt++;
        }
        System.out.println(cnt);
    }
}